packages/rocketchat-lib/startup/defaultRoomTypes.js

```diff
@@ -102,6 +103,10 @@ RocketChat.roomTypes.add('p', 40, {
 		const user = Meteor.user();
 		const preferences = (user && user.settings && user.settings.preferences && user.settings.preferences) || {};
 		return !preferences.roomsListExhibitionMode || ['unread', 'category'].includes(preferences.roomsListExhibitionMode) && !preferences.mergeChannels && RocketChat.authz.hasAllPermission('view-p-room');
+	},
+
+	showJoinLink() {
+		return true;
```

Comment:
If this is going to just be true with no logic do we really need this?

Reply:
so, for private groups there is no logic because it will always show a "Room not found" message and will not render the "join link". unless it's a TCA private group and you can see it (because you have the needed token), in this case it will always show the join link =)